Flaky-test: V5ProducerAccessModeTest > testExclusiveRejectsSecondProducer

### Search before reporting

- [x] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/25387350198/job/74454796185?pr=25675#step:11:249

### Exception stacktrace

```text
Gradle suite > Gradle test > org.apache.pulsar.client.api.v5.V5ProducerAccessModeTest > testExclusiveRejectsSecondProducer FAILED
      java.lang.AssertionError: a second EXCLUSIVE producer should not be able to attach
          at org.testng.Assert.fail(Assert.java:111)
          at org.apache.pulsar.client.api.v5.V5ProducerAccessModeTest.testExclusiveRejectsSecondProducer(V5ProducerAccessModeTest.java:92)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-client-v5/src/main/java/org/apache/pulsar/client/impl/v5/ScalableTopicProducer.java
+++ b/pulsar-client-v5/src/main/java/org/apache/pulsar/client/impl/v5/ScalableTopicProducer.java
@@ -389,7 +389,7 @@ final class ScalableTopicProducer<T> implements Producer<T>, DagWatchClient.Layo
         // that segment will surface the error via the normal PulsarClientException
         // path. (The initial-create path uses {@link #eagerAttachInitialAsync} for
         // strict claim.)
-        if (requiresExclusiveAttach()) {
+        if (requiresExclusiveAttach() && oldLayout != null) {
             CompletableFuture.runAsync(() -> {
                 for (var seg : newLayout.activeSegments()) {
                     if (segmentProducers.containsKey(seg.segmentId())) {
@@ -449,9 +449,6 @@ final class ScalableTopicProducer<T> implements Producer<T>, DagWatchClient.Layo
         }
         return CompletableFuture.runAsync(() -> {
             for (var seg : activeSegments) {
-                if (segmentProducers.containsKey(seg.segmentId())) {
-                    continue;
-                }
                 try {
                     getOrCreateSegmentProducer(seg.segmentId());
                 } catch (PulsarClientException e) {

